test/models/test_fully_bayesian.py

```diff
@@ -656,6 +666,77 @@ def test_custom_pyro_model(self):
                     atol=5e-4,
                 )
 
+    def test_condition_on_observation(self):
+
+        num_models = 3
+        num_cond = 2
+        for infer_noise, dtype in itertools.product(
+            (True,), (torch.float, torch.double)
+        ):
```

Comment:
Looks like you intending to test `infer_noise=False` as well here? You could also just test one option for each.
```suggestion
        for infer_noise, dtype in ((True, torch.float), (False, torch.double)):
```